[WIP] Don't overwrite es_plugins_reinstall but set es_plugins_install instead

barryib opened the pull request:
When using tags or setting explicitly ``es_plugins_reinstall`` to true in vars, this last is overwrite with a ``set_fact``.

With this PR, we don't overwrite ``es_plugins_reinstall``, but we ``set es_plugins_install`` instead to switch the plugin install.

Here is a related issue #147. See [this](https://github.com/elastic/ansible-elasticsearch/issues/147#issuecomment-245434439) comment for details.

elasticsearch-release commented:
Jenkins standing by to test this. If you aren't a maintainer, you can ignore this comment. Someone with commit access, please review this and clear it for Jenkins to run.

bhcopeland commented:
+1 works here.

Although, when I set `es_plugins_reinstall: false` the plugins are still reinstalled.

gingerwizard commented:
@barryib if some explicitly sets es_plugins_install: false then this will potentially still result in plugin install.  Maybe make the setting of es_plugins_install conditional on it not being set explictly?

We could of actually done that for es_plguins_reinstall...which might be clearer?

gingerwizard commented:
jenkins, test it

barryib commented:
@gingerwizard @bhcopeland I that this is a regression introduced by this commit 53535241370db11c77fac379f547f65fb49b1a13. See this [line](https://github.com/elastic/ansible-elasticsearch/commit/53535241370db11c77fac379f547f65fb49b1a13#diff-3f419eb05dd0502a4e53b27340029a92L5).

In that previous version, we only set ``es_plugins_reinstall`` to true when one of these following statements are true:
- ``elasticsearch_install_from_repo.changed`` - for plugins update
- ``elasticsearch_install_from_package.changed`` - for plugins update
- ``es_plugins`` is none - for plugins removal

In other way, you can explicitly set ``es_plugins_reinstall`` to false to avoid plugins reinstallation if nothing need to be updated.

gingerwizard commented:
@barryib agreed i think we just revert this...ill close this and do a push now

barryib commented:
@gingerwizard Thanks.
